### ISSUE TITLE
Include suggestions for using git with signac

### ISSUE THREAD

**Issue by atravitz** [glotzerlab/signac-docs#100]
We should acknowledge git in our documentation somewhere

### Issue Description
Common questions I've gotten from new users (and questions I've asked myself) are "what should I put in my git ignore", "should I commit my workspace to git", and "can I keep track of my git hash in the job document". 

### Proposed change

I think we should include some best practices that we all tend to use with git, such as *not* including the workspace in your git repo. Maybe this can go in FAQs?

### Additional context

An additional topic for discussion here is: does anyone have a reliable way of tracking what code version they used to generate data for a specific job? Tracking the git hash is one way, but I'm interested to see if anyone has thought of other ideas.

**Comment by vyasr** [comment]
Regarding the Additional Context, this is what glotzerlab/signac-flow#189 is meant to help with, but yes git hashes are probably one of the best options.

**Comment by bdice** [comment]
Here's what I put in my `.gitignore`. This probably isn't a complete list, make suggestions if I'm overlooking something.
```
.bundles
.signac_shell_history
.signac_sp_cache.json.gz
signac.rc
signac_data.h5
signac_project_document.json
workspace
```

**Comment by csadorf** [comment]
It's really time that we long-term move towards `.signac/` for all those files cluttering the project root directory, huh? :)

**Comment by vyasr** [comment]
Yup, definitely think that's the right move long term! Documented at glotzerlab/signac#197 and glotzerlab/signac-flow#347
